src/plugins/discover/public/application/angular/discover.js

```diff
@@ -979,20 +707,11 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
   };
 
   $scope.updateDataSource = () => {
-    const { indexPattern, searchSource } = $scope;
-    searchSource
-      .setField('index', $scope.indexPattern)
-      .setField('size', $scope.opts.sampleSize)
-      .setField(
-        'sort',
-        getSortForSearchSource(
-          $scope.state.sort,
-          indexPattern,
-          config.get(SORT_DEFAULT_ORDER_SETTING)
-        )
-      )
-      .setField('query', data.query.queryString.getQuery() || null)
-      .setField('filter', filterManager.getFilters());
```

Comment:
extracted to helper function: https://github.com/elastic/kibana/blob/5a1f3ef6f740315b7b41013f840a36b48cd68c66/src/plugins/discover/public/application/helpers/update_search_source.ts#L29